### PR TITLE
Fixforth and msr

### PR DESCRIPTION
**Pull request by rminnich**
I think this makes the other msr PR unnecessary. 

I started to rip the forth out of msr, but ... it's just too handy. I figure if I can make the experience less awful, it may be nice to leave it in.

### CODE REVIEW

**Comment by rminnich** [comment]
I still need to change the docs.

Suggestions on better ways to do things are most welcome.
I wanted to use constraint.Integer but don't understand how.

Long term, I want to support bignum.Rat on the stack, instead of uint64, but that's for later if we ever need it.

Some of this messing around belongs in the forth package I suspect. But I doubt anyone cares but me.

But it is HANDY to be able to do stuff like this
msr * 0x3a rd * 0x3c rd
That said, this is still not quite right. I don't think the cpu and msrs should be popped off when used.
Then you could do stuff like
msr * 0x3a rd 1 and wr

to easily clear all bits in all msrs in all cores.

**Comment by rminnich** [comment]
note that, also, the unpleasant behavior was fixed. Due to an old bug, panics were leaking outside the forth package. Yuck.

**Comment by rminnich** [comment]
ok, this is a serious improvement to this admittedly craptastic command. I just to get so mad at it when I used it, now I'm finding it very congenial.

File: pkg/forth/forth.go [hunk 1]
```diff
@@ -183,13 +184,18 @@ func (f *stack) Empty() bool {
...
+	Debug("pkg/forth:e is %v:%T", e, e)
 	if e != nil {
-		if _, ok := e.(runtime.Error); ok {
-			Debug("errRecover panics with a runtime error")
+		switch err := e.(type) {
+		case runtime.Error:
+			Debug("pkg/forth:errRecover panics with a runtime error")
 			panic(e)
+		case error:
+			*errp = fmt.Errorf("%w", err)
```
Comment (binjip978) [inline pkg/forth/forth.go:194]: maybe `*errp = err`?

File: pkg/forth/forth.go [hunk 2]
```diff
@@ -183,13 +184,18 @@ func (f *stack) Empty() bool {
...
+		default:
+			*errp = fmt.Errorf("pkg/forth:%v", err)
```
Comment (binjip978) [inline pkg/forth/forth.go:196]: and %v -> %w here

**Review by binjip978 — DISMISSED** [review]

File: pkg/forth/forth.go (hunk 2 above)
Comment (rminnich) [reply pkg/forth/forth.go:196]: can't do that here, it's not an error, it's any

File: pkg/forth/forth.go (hunk 1 above)
Comment (rminnich) [reply pkg/forth/forth.go:194]: done

**Review by binjip978 — APPROVED** [review]

**Pull request merged**